src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Server/ServerInfo.cs

```diff
@@ -28,7 +28,14 @@ public class ServerInfo : SqlObject
         public string? ServiceTier { get; set; }
         public int? StorageSpaceUsageInMB { get; set; }
         public string Version { get; set; }
-        public int MaxServerMemory { get; set; }
-        public int MinServerMemory { get; set; }
+        public ServerProperty MaxServerMemory { get; set; }
+        public ServerProperty MinServerMemory { get; set; }
+    }
+
+    public class ServerProperty
```

Comment:
Maybe `NumericServerProperty`? Since we can have other types of server properties too